src/typing.py

```diff
@@ -1494,7 +1495,10 @@ def get_type_hints(obj, globalns=None, localns=None):
     if getattr(obj, '__no_type_check__', None):
         return {}
     if globalns is None:
-        globalns = getattr(obj, '__globals__', {})
+        if isinstance(obj, type):
+            globalns = vars(sys.modules[obj.__module__])
```

Comment:
While you at it maybe you can fix same way also the logic below when ``__mro__`` is traversed
(I could imagine that base classes are from different modules). Or will it be too hard? If not, I would add also a  test for such scenario.

Reply:
Yeah, I have a few ideas more but it's annoying that we don't ship `ann_module.py` et al. but always refer to the one in `test` for Python 3.5+. This makes it hard to iterate on them.

Let me see what I can do.

Reply:
Done!